resource-managers/kubernetes/integration-tests/dev/dev-run-integration-tests.sh

```diff
@@ -119,4 +139,15 @@ then
   properties=( ${properties[@]} -Dtest.exclude.tags=$EXCLUDE_TAGS )
 fi
 
+BASE_IMAGE_NAME=${BASE_IMAGE_NAME:-spark}
+JVM_IMAGE_NAME=${JVM_IMAGE_NAME:-${BASE_IMAGE_NAME}}
+PYTHON_IMAGE_NAME=${PYTHON_IMAGE_NAME:-${BASE_IMAGE_NAME}-py}
+R_IMAGE_NAME=${R_IMAGE_NAME:-${R_IMAGE_NAME}-r}
+
+properties+=(
```

Comment:
it seems this `roperties+=(...` can not work, I have added a echo statement before the `mvn` command, it shows the `properties` didn't contain the image properties, see following log message:

```
+ properties=(${properties[@]} -Dtest.exclude.tags=$EXCLUDE_TAGS)
+ BASE_IMAGE_NAME=spark
+ JVM_IMAGE_NAME=spark
+ PYTHON_IMAGE_NAME=spark-py
+ R_IMAGE_NAME=-r
+ properties+=(-Dspark.kubernetes.test.jvmImage=$JVM_IMAGE_NAME -Dspark.kubernetes.test.pythonImage=$PYTHON_IMAGE_NAME -Dspark.kubernetes.test.rImage=$R_IMAGE_NAME)
+ echo -Dspark.kubernetes.test.sparkTgz=/home/zuul/src/github.com/liu-sheng/spark/spark-3.0.0-SNAPSHOT-bin-2.7.4.tgz
```

Reply:
@liu-sheng What did you echo?  An `echo $properties` will only echo the first item in the array, an `echo ${properties[@]}` echos the entire array.  It looks like you only echo'd the first item in the array

Note that you can invoke the script with `sh -x dev/dev-run-integration-tests.sh` to get detailed Bash debug output.

Reply:
> @liu-sheng What did you echo? An `echo $properties` will only echo the first item in the array, an `echo ${properties[@]}` echos the entire array. It looks like you only echo'd the first item in the array
> 
> Note that you can invoke the script with `sh -x dev/dev-run-integration-tests.sh` to get detailed Bash debug output.

Oh, my fault, let me try, thank you for teaching me :)

Reply:
BTW, the L145 `R_IMAGE_NAME=${R_IMAGE_NAME:-${R_IMAGE_NAME}-r}`, is the default `R_IMAGE_NAME` var name should be `BASE_IMAGE_NAME` ? if I do not specify the R image with `--r-image-name`, the R image name will be `-r` as default.

Reply:
Oops, then that's a bug that made it past the review.  Opened PR #24449

Reply:
@liu-sheng This should now be fixed in latest master

Reply:
@rvesse thank you :)